src/aggregate/aggregate_exec.c

```diff
@@ -190,8 +190,7 @@ void sendChunk(AREQ *req, RedisModuleCtx *outctx, size_t limit) {
 
   rc = rp->Next(rp, &r);
   long resultsLen = REDISMODULE_POSTPONED_ARRAY_LEN;
-  if (rc == RS_RESULT_TIMEDOUT && !(req->reqflags & QEXEC_F_IS_CURSOR) && !IsProfile(req) &&
-      RSGlobalConfig.timeoutPolicy == TimeoutPolicy_Fail) {
+  if (rc == RS_RESULT_TIMEDOUT && !(req->reqflags & QEXEC_F_IS_CURSOR) && !IsProfile(req)) {
```

Comment:
Can you explain this change, only if `RSGlobalConfig.timeoutPolicy == TimeoutPolicy_Fail` you want to return a single results which is the timeout no?